app/views/wishlists_for_guests/_item.html.erb

```diff
@@ -5,8 +5,17 @@
 
       <p><%= item.description %></p>
 
-      <%= link_to "#", class: "btn btn-default btn-sm" do %>
-        <span class="glyphicon glyphicon-gift"></span> I'll buy it!
+      <%= form_for(@reservation, url: wishlist_item_reservations_path(item.wishlist_id, item.id)) do |f| %>
+        <%= button_tag(type: 'submit', class: "btn btn-default btn-sm submit", disabled: !item.reservation.nil?) do %>
+
+          <span class="glyphicon glyphicon-gift"></span>
+
+          <% if item.reservation.nil? %>
```

Comment:
I'd add a helper method `reserved?` in the `Item` model. It'll simply look better.